Fix unused function warning in DocumentLoader.

mokjasmine opened the pull request:
#### e8e6c7f9748df4cb53685bc4dc67b8595e11eeed
<pre>
Fix unused function warning in DocumentLoader.
<a href="https://bugs.webkit.org/show_bug.cgi?id=253589">https://bugs.webkit.org/show_bug.cgi?id=253589</a>

Reviewed by Fujii Hironori.

Move `isSandboxingAllowingServiceWorkerFetchHandling` into
`DocumentLoader::loadMainResource()` to fix unused function
warning when `SERVICE_WORKER` is OFF.

* Source/WebCore/loader/DocumentLoader.cpp:

Canonical link: <a href="https://commits.webkit.org/261485@main">https://commits.webkit.org/261485@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/17b6f0dda8007169426d26e7eb1396afb719074b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111783 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20912 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/399 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3559 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120493 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115842 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22271 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11987 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/3305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117548 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16542 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99689 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104788 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98494 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/265 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/45513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13378 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/266 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/92337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13875 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9699 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19319 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52268 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7993 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15860 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/loader/DocumentLoader.cpp:
```diff
@@ -2165,7 +2165,7 @@ void DocumentLoader::unregisterReservedServiceWorkerClient()
 }
 
-static bool isSandboxingAllowingServiceWorkerFetchHandling(SandboxFlags flags)
+static UNUSED_FUNCTION bool isSandboxingAllowingServiceWorkerFetchHandling(SandboxFlags flags)
```
Use `#if ENABLE(SERVICE_WORKER)`.

fujii replied:
Or, you can move this into DocumentLoader::loadMainResource.
> auto isSandboxingAllowingServiceWorkerFetchHandling = [](SandboxFlags flags) {
>     return !(flags & SandboxOrigin) && !(flags & SandboxScripts);
> }

mokjasmine replied:
Ok sure. Can you tell me why? I'm curious about the when to use one method over the other.

fujii replied:
Everyone does so in such a case. I just follow the WebKit style. And, I don't know when to use `UNUSED_FUNCTION`.

mokjasmine committed:
Fix unused function warning in DocumentLoader.
https://bugs.webkit.org/show_bug.cgi?id=253589

Reviewed by Fujii Hironori.

Move `isSandboxingAllowingServiceWorkerFetchHandling` into
`DocumentLoader::loadMainResource()` to fix unused function
warning when `SERVICE_WORKER` is OFF.

* Source/WebCore/loader/DocumentLoader.cpp:

Canonical link: https://commits.webkit.org/261485@main
